### PR TITLE
add :cinv(), which does 1/x

### PR DESCRIPTION
**Pull request by hughperkins**
Add :cinv(), which does 1/x.  Use like `:cinv()`, or `torch.cinv(...)`

Addresses https://github.com/torch/torch7/issues/168

### CODE REVIEW

**Comment by hughperkins** [comment]
By the way, I also renamed the c function `sigmoid` to `TH_sigmoid`, to avoid possible conflicts, and made it `static inline`, and remove THMath.c.

**Comment by jonathantompson** [comment]
Suggestion: maybe rename to cinv?  Otherwise people are going to (naturally) think that this calculates a matrix inverse (we defined cmul for the same reason).

**Comment by hughperkins** [comment]
Ah, good point.

**Comment by hughperkins** [comment]
ah, yes, `cinv` is perfect really, ie 'per-element inverse'.

**Comment by jonathantompson** [comment]
Cool.  Thanks for the PR!  You've definitely be on fire lately with all your contributions :-)

File: lib/TH/generic/THTensorMath.c [hunk 1]
```diff
@@ -1995,6 +1995,7 @@ LAB_IMPLEMENT_BASIC_FUNCTION(floor,floor)
 LAB_IMPLEMENT_BASIC_FUNCTION(round,round)
 LAB_IMPLEMENT_BASIC_FUNCTION(abs,fabs)
 LAB_IMPLEMENT_BASIC_FUNCTION(neg,-)
+LAB_IMPLEMENT_BASIC_FUNCTION(inv,TH_inv)
```
Comment (fmassa) [inline lib/TH/generic/THTensorMath.c]: Just wondering, shouldn't we try to be consistent in here ? [In your cutorch PR](https://github.com/torch/cutorch/pull/306/files#diff-13ee6e0e8f659bdf5f14261f7bee76a5R54), you used `1.0f /`.

Comment (hughperkins) [inline lib/TH/generic/THTensorMath.c]: Hmmm, yes.  That does make sense.  I'll see if that works.

Comment (hughperkins) [inline lib/TH/generic/THTensorMath.c]: Addressed

**Comment by hughperkins** [comment]
Travis didnt like that.  Random chance?  Should change back to using `TH_cinv`, instead of `1.0 / ` in the macro?  I'll try pushing some tiny change, nad see what happens.

**Comment by soumith** [comment]
Thanks Hugh!

**Pull request merged**